### PR TITLE
[Improve][Connector-V2][Hive] Hive Sink Support msck partitions

### LINKED ISSUE #3132: [Improve][Connector-V2][Hive] Support for automatic partition repair
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Hive connector supports automatic partition repair

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

Hive connector supports automatic partition repair

close #3132 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
local test:

image

image

integration test:

image

image

File: seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/commit/HiveSinkAggregatedCommitter.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        List<FileAggregatedCommitInfo> errorCommitInfos = super.commit(aggregatedCommitInfos);
+        if (errorCommitInfos.isEmpty()) {
+            for (FileAggregatedCommitInfo aggregatedCommitInfo : aggregatedCommitInfos) {
+                Map<String, List<String>> partitionDirAndValuesMap = aggregatedCommitInfo.getPartitionDirAndValuesMap();
+                List<String> partitions = partitionDirAndValuesMap.keySet().stream()
+                        .map(partition -> partition.replaceAll("\\\\", "/"))
+                        .collect(Collectors.toList());
+                try {
+                    hiveMetaStore.addPartitions(dbName, tableName, partitions);
+                    log.info("Msck repair these partitions [{}]", String.join(",", partitions));
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/commit/HiveSinkAggregatedCommitter.java]:
```suggestion
                    log.info("Msck repair these partitions {}", partitions);
```

update elsewhere

**Review by hailin0 — COMMENTED** [review]
LGTM

File: seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/commit/HiveSinkAggregatedCommitter.java (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/commit/HiveSinkAggregatedCommitter.java]: good advice！

**Review by CalvinKirs — DISMISSED** [review]

**Review by TaoZex — DISMISSED** [review]

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**